core/src/avm2/globals/flash/display/bitmap.rs

```diff
@@ -45,20 +48,60 @@ pub fn instance_init<'gc>(
                 //need to create bitmap data right away, since all AVM2 bitmaps
                 //hold bitmap data.
 
-                if let Some(bd_class) = bitmap.avm2_bitmapdata_class() {
-                    let bd_object = bd_class.construct(activation, &[])?;
+                let bd_object = if let Some(bd_class) = bitmap.avm2_bitmapdata_class() {
+                    bd_class.construct(activation, &[])?
```

Comment:
if we did the `has_class_in_chain` check here, would we still need the `BitmapClass` enum?

Reply:
Nope. Do you want me to remove the enum?

Reply:
Feels simpler to me this way.